extensions/wikia/UserRenameTool/SpecialRenameuser_body.php

```diff
@@ -32,99 +31,179 @@ public function execute( $par ) {
 		$this->setHeaders();
 		$this->checkPermissions();
 
-		$out = $this->getOutput();
-		$this->getOutput()->addScript(
-			Html::linkedScript(
-				AssetsManager::getInstance()->getOneCommonURL( '/extensions/wikia/UserRenameTool/js/NewUsernameUrlEncoder.js' )
-			)
-		);
+		$username = \User::isValidUsername( $par ) ? $par : null;
+		$requestorUser = $this->getUser();
+
+		if ( !$requestorUser->isAllowed( 'renameuser' ) ) {
+			throw new PermissionsError( 'renameuser' );
+		}
+
+		if ( $username && !$requestorUser->isAllowed( 'renameanotheruser' ) ) {
+			throw new PermissionsError( 'renameanotheruser' );
+		}
 
 		if ( wfReadOnly() || !$wgStatsDBEnabled ) {
-			$out->readOnlyPage();
+			$this->getOutput()->readOnlyPage();
+
 			return;
 		}
 
-		// Get the request data
-		$request = $this->getRequest();
-		$oldUsername = $request->getText( 'oldusername', $par );
-		$newUsername = $request->getText( 'newusername' );
-		$reason = $request->getText( 'reason' );
-		$token = $this->getUser()->getEditToken();
-		$notifyRenamed = $request->getBool( 'notify_renamed', false );
-		$confirmAction = false;
+		// TODO: remove after QA tests
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this "TODO" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)

[project-id]: wikia:app